### PR TITLE
Add Ability to load Certificates from Data Resources/Secrets

### PR DESCRIPTION
**Pull request by davidcollom**
The current implementation for TLS requires that the certificates exist on disk - however, if the only access you have to these secrets is via an external secret manager (say SecretManager, Vault etc).

I did think of creating two new parameters but it felt like it was easier to discover if someone had used a string of a cert instead of a file path.

### CODE REVIEW

**Comment by petoju** [comment]
Could you please add documentation to the respective option? Documentation is in the folder `website`

**Comment by davidcollom** [comment]
Yep! I can add those in the next hour or so! 👍

File: website/docs/index.html.markdown [hunk 1]
```diff
@@ -79,6 +79,66 @@ provider "mysql" {
...
+    client_key  = var.mysql_tls_client_key
+  }
+}
+```
+
+**Note** It it is _strongly_ recommended to ensure that these values/variables are marked as sensitive
+
+
+
+
```
Comment (davidcollom) [inline website/docs/index.html.markdown:141]:
```suggestion
```

File: website/docs/index.html.markdown [hunk 2]
```diff
@@ -79,6 +79,63 @@ provider "mysql" {
 }
 ```
 
+TLS Configuration can also be sourced from secrets such as Google Secret Manager or variables like so:
```
Comment (petoju) [inline website/docs/index.html.markdown:82]: This does not say a lot. Anything can be in that secrets.

What I imagine: when I open
https://registry.terraform.io/providers/petoju/mysql/latest/docs#ca_cert
(sourced from here), I will see, that ca_cert can be a filename or a text certificate starting with `-----BEGIN`. Similar with the key.

Once anyone knowing terraform spots that those may be just file contents, he can easily use google secrets manager or basically anything.

Comment (petoju) [reply website/docs/index.html.markdown]: Np, this looks better!

**Review by petoju — APPROVED** [review]

**Comment by petoju** [comment]
Release in v3.0.52

**Pull request merged**